13167 docs for connector builder new UI schemaform rollout (#63329)
airbytehq/airbyte: docs/platform/connector-development/connector-builder-ui/async-streams.md

  - Set **Timeout** to: `timeout`
- In the **Download Target Extractor** section:
  - Set the **Field Path** to: `urls`
- **Polling Job Timeout**: Set an appropriate timeout value (e.g., `PT30M` for 30 minutes)

### 3. Download Tab

The Download tab configures how to retrieve the results once the job is complete. This tab provides comprehensive configuration options for processing the downloaded data.

![Download Tab](./assets/connector_builder_async_download_tab.png)

#### Key Components:

> - **URL** (`retriever.download_requester.url`): The full URL for downloading results. Use the `{{ download_target }}` variable to reference the value extracted by the Download Target Extractor (required)
> - **HTTP Method** (`retriever.download_requester.http_method`): HTTP method for downloading, typically GET but can vary by API
> - **HTTP Response Format** (`retriever.download_decoder`): Format of the downloaded data (JSON, CSV, XML, etc.)
> - **Record Selector** (`retriever.record_selector`): Configuration for identifying and extracting individual records from the response, including the extractor field path
> - **Primary Key** (`primary_key`): Unique identifier field(s) for records in the stream
> - **Request Parameters** (`retriever.download_requester.request_parameters`): Query parameters for download requests
> - **Request Headers** (`retriever.download_requester.request_headers`): HTTP headers for download requests
> - **Request Body** (`retriever.download_requester.request_body`): Request body for download requests (if needed)
> - **Authentication** (`retriever.download_requester.authenticator`): Authentication method for download requests
> - **Download Paginator** (`retriever.download_paginator`): Pagination configuration if the download response is paginated
> - **Transformations** (`transformations`): Data transformation rules to apply to downloaded records
> - **Error Handler** (`retriever.download_requester.error_handler`): Error handling configuration for download requests

Reviewer: ```suggestion
- **API Endpoint URL** (required): The full URL for downloading results. Use the `{{ download_target }}` variable to reference the value extracted by the Download Target Extractor
- **HTTP Method**: HTTP method for downloading, typically GET but can vary by API
- **Download HTTP Response Format**: Format of the downloaded data (JSON, CSV, XML, etc.)
- **Record Selector**: Configuration for identifying and extracting individual records from the response, including the extractor field path
- **Primary Key**: Unique identifier field(s) for records in the stream
- **Authenticator**: Authentication method for download requests
- **Query Parameters**: Query parameters for download requests
- **Request Headers**: HTTP headers for download requests
- **Request Body**: Request body for download requests
- **Download Paginator**: Pagination configuration if the download response is paginated
- **Transformations**: Data transformation rules to apply to downloaded records
- **Error Handler**: Error handling configuration for download requests
```

   - Verify that status is at the expected place with the expected values in the polling response 
   - Ensure the download target is correctly extracted by checking the main Request tab URL

> 2. **Configure Status Mapping Properly**: Include all possible API status values in your status mapping. The schema-driven forms will validate that you've provided mappings for all required states.

Reviewer: ```suggestion
2. **Configure Status Mapping Properly**: Include all possible API status values in your status mapping.
```

3. **Use the Right HTTP Response Format**: Make sure you select the correct HTTP Response Format (JSON, CSV, XML, etc.) for both creation and download phases.

4. **Configure Appropriate Timeouts**: Set reasonable values for `polling_job_timeout` to avoid indefinite waiting for job completion.

5. **Handle Rate Limits**: Many APIs limit how frequently you can poll for job status. Configure appropriate error handling and retry strategies.

> 6. **Test Thoroughly**: Use the comprehensive testing features to verify each phase of the asynchronous process works correctly.

Reviewer: ```suggestion
6. **Test Thoroughly**: Test the stream verify each phase of the asynchronous process works correctly.
```

> 7. **Leverage Schema Validation**: The schema-driven forms provide real-time validation feedback to help you catch configuration errors early.

Reviewer: ```suggestion
```

> Remember that asynchronous streams often take longer to test than synchronous streams, especially if the API takes time to process jobs. The schema-driven UI provides comprehensive feedback during testing to help you understand what's happening at each stage.     

Reviewer: ```suggestion
Remember that asynchronous streams often take longer to test than synchronous streams, especially if the API takes time to process jobs.
```
